manifests/plugin/python.pp

```diff
@@ -1,45 +1,51 @@
 # See http://collectd.org/documentation/manpages/collectd.conf.5.shtml#plugin_python
-define collectd::plugin::python (
-  $modulepath,
-  $module,
-  $script_source,
-  $ensure = present,
-  $config = {},
-  $order = '10',
+class collectd::plugin::python (
+  $modulepath = '/usr/lib/collectd/python',
+  $ensure     = present,
+  $modules    = {},
+  $order      = '10',
+  $interval   = undef,
 ) {
   include collectd::params
 
-  validate_hash($config)
+  validate_hash($modules)
 
   $conf_dir = $collectd::params::plugin_conf_dir
 
-  # This is deprecated file naming ensuring old style file removed, and should be removed in next major relese
-  file { "${name}.load-deprecated":
-    ensure => absent,
-    path   => "${conf_dir}/${name}.conf",
+  collectd::plugin {'python':
+    ensure   => $ensure,
+    interval => $interval,
+    order    => $order,
   }
-  # End deprecation
-
-  file {
-    "${name}.load":
-      ensure  => $ensure,
-      path    => "${conf_dir}/${order}-${name}.conf",
-      owner   => 'root',
-      group   => $collectd::params::root_group,
-      mode    => '0640',
-      content => template('collectd/python.conf.erb'),
-      notify  => Service['collectd'],
+
+  # should be loaded after global plugin configuration
+  $python_conf = "${collectd::params::plugin_conf_dir}/11-python-config.conf"
```

Comment:
You don't need to hardcode '11'. all LoadPLugin directives will be in a file starting with a number - this config file should be called python-config.conf and will be loaded afterwards anyway